core/parentchain/light-client/src/concurrent_access.rs

```diff
@@ -26,32 +26,29 @@ use std::sync::RwLock;
 
 use crate::{
 	error::{Error, Result},
-	LightClientState, Validator as ValidatorTrait,
+	ExtrinsicSender as ExtrinsicSenderTrait, LightClientState, LightValidationState,
+	Validator as ValidatorTrait,
 };
 use finality_grandpa::BlockNumberOps;
+use itp_ocall_api::EnclaveOnChainOCallApi;
 use itp_sgx_io::StaticSealedIO;
-use lazy_static::lazy_static;
 use sp_runtime::traits::{Block as ParentchainBlockTrait, NumberFor};
 use std::marker::PhantomData;
 
-lazy_static! {
-	// As long as we have a file backend, we use this 'dummy' lock,
-	// which guards against concurrent read/write access.
-	pub static ref VALIDATOR_LOCK: RwLock<()> = Default::default();
-}
-
 /// Retrieve an exclusive lock on a validator for either read or write access.
 ///
 /// In order to hide the whole locks mechanics, we provide an interface that allows executing
 /// either a mutating, or a non-mutating function on the validator.
 /// The reason we have this additional wrapper around `SealedIO`, is that we need
 /// to guard against concurrent access by using RWLocks (which `SealedIO` does not do).
-pub trait ValidatorAccess<ParentchainBlock>
+pub trait ValidatorAccess<ParentchainBlock, OCallApi: EnclaveOnChainOCallApi>
```

Comment:
Please put the trait bound below in the `where` statement, like the rest of the bounds.